BudgetServiceTests/BudgetCalculationsTests.swift

```diff
@@ -0,0 +1,147 @@
+//
+//  BudgetCalculationsTests.swift
+//  BudgetServiceTests
+//
+//  Created by antuan.khoanh on 15/07/2022.
+//
+
+import XCTest
+@testable import bonsai
+
+class BudgetCalculationsTests: XCTestCase {
+
+    func testBudgetMoneyCanSpendToday() throws {
+        // g
+        
+        let budgetCalculations = makeSUT()
+        
+        let input: [(amount: NSDecimalNumber, days: Int64)] = [
+            (amount: 90, days: 30),
+            (amount: 120, days: 30),
+            (amount: 30, days: 30),
+            (amount: 150, days: 30),
+            (amount: 160, days: 30),
+            (amount: 880, days: 29)
+        ]
+        
+        // w
+        
+        let results: [NSDecimalNumber] = input.map {
+            budgetCalculations.calculateMoneyCanSpendDaily(
+                currentAmount: $0.amount,
+                periodDays: $0.days
+            )
+        }
+        
+        // t
+        
+        let expectations: [NSDecimalNumber] = [
+            .init(value: 3),
+            .init(value: 4),
+            .init(value: 1),
+            .init(value: 5),
+            .init(value: 5.33),
+            .init(value: 30.34)
+        ]
+        
+        for i in 0..<results.count {
+            XCTAssertNotNil(expectations[i])
+            let res = results[i]
+            let exp = expectations[i]
+            XCTAssert(res == exp, "res=\(res); exp=\(exp)")
```

Comment:
Or use this:
`XCTAssertEqual(res, exp)`
Error output:
`XCTAssertEqual failed: ("30.34") is not equal to ("30.3")`

Reply:
+fixed